kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/BigQueryWriteSettingsBuilder.java

```diff
@@ -0,0 +1,36 @@
+package com.wepay.kafka.connect.bigquery.write.storageApi;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.bigquery.storage.v1.BigQueryWriteSettings;
+import com.wepay.kafka.connect.bigquery.GcpClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+
+public class BigQueryWriteSettingsBuilder extends GcpClientBuilder<BigQueryWriteSettings> {
+
+    private static final Logger logger = LoggerFactory.getLogger(BigQueryWriteSettingsBuilder.class);
+
+    @Override
+    protected BigQueryWriteSettings doBuild(String project, GoogleCredentials credentials, HeaderProvider userAgent) {
+        BigQueryWriteSettings.Builder builder = BigQueryWriteSettings.newBuilder()
+                .setQuotaProjectId(project)
+                .setHeaderProvider(userAgent);
+
+
+        if (credentials != null) {
+            builder.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
+        } else {
+            logger.info("Attempting to access GCS without authentication");
```

Comment:
Is this a usual case or we should log this at WARN level? Do users attempts to access GCS without authentication - also is it allowed?

Reply:
It should be warn, will update. No, users cannot authenticate like this.